app/assets/javascripts/application.js.erb

```diff
@@ -31,48 +31,56 @@ function human_time(seconds_total) {
   return `${hours_str}:${minutes_str}:${seconds_str}`;
 }
 
+function handleErrors(response) {
+    if (!response.ok) {
+        table.processing(false);
+        throw Error(response.statusText);
+    }
+    return response;
+}
+
+function checkResponse(type) {
```

Comment:
After looking at the code, it might be better to inline the `checkResponse` function back into the body of the fetch_table_data function and instead place a comment above the block explaining what `result.value` is (i.e. either an array of job objects or an error object).